src/Common/src/System/IO/PathInternal.cs

```diff
@@ -23,5 +25,67 @@ internal static void CheckInvalidPathChars(string path, bool checkAdditional = f
             if (PathInternal.HasIllegalCharacters(path, checkAdditional))
                 throw new ArgumentException(SR.Argument_InvalidPathChars, "path");
         }
+
+
+        /// <summary>
+        /// Returns true if the given StringBuilder starts with the given value.
+        /// </summary>
+        /// <param name="value">The string to compare against the start of the StringBuilder.</param>
+        internal static bool StartsWithOrdinal(this StringBuilder builder, string value)
+        {
+            if (value == null || builder.Length < value.Length)
+                return false;
+
+            for (int i = 0; i < value.Length; i++)
+            {
+                if (builder[i] != value[i]) return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Returns true if the given string starts with the given value.
+        /// </summary>
+        /// <param name="value">The string to compare against the start of the source string.</param>
+        internal static bool StartsWithOrdinal(this string source, string value)
+        {
```

Comment:
It would differ in the behavior around null. I could change it to check null then call the other- thoughts?